localstack/packages/api.py

```diff
@@ -59,6 +60,7 @@ def __init__(self, name: str, version: str):
         """
         self.name = name
         self.version = version
+        self.lock = threading.RLock()
```

Comment:
As you mentioned in the PR description, this lock behavior only works because the Package's `get_installer` method caches the instances per version.
As this now becomes a necessity (and is not just there for performance reasons), it would be nice to add this to the docstring of the `get_installer` function (that it needs to be ensured that there is only a single instance of an installer class per version).

Reply:
Added in 726b0623 :+1: